client/src/components/food/FoodItemsSelected/index.js

```diff
@@ -0,0 +1,113 @@
+import React, { useEffect, useState } from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import axios from 'axios';
+import PropTypes from 'prop-types';
+import { useHistory } from 'react-router-dom';
+import FoodItems from '../FoodItems';
+import Container from '../../Container';
+import Button from '../../Button';
+import { MyFood } from '../../../Utils/constant';
+import Loading from '../../Loading';
+import CustomErrorMessage from '../CustomErrorMessage';
+
+const { number } = PropTypes;
+
+const useStyle = makeStyles((theme) => ({
+  button: {
+    color: theme.customColors.white,
+    fontSize: '20px',
+    textTransform: 'capitalize',
+  },
+  container: {
+    width: '300px',
+  },
+  totalCalories: {
+    fontSize: '20px',
+    fontWeight: '800',
+  },
+}));
+
+function FoodItemsSelected(props) {
+  const { foodCategoryId } = props;
+  const history = useHistory();
+  const [foodArray, setFoodArray] = useState([]);
+  const [totalCalories, setTotalCalories] = useState(0);
+  const [showLoading, setShowLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  useEffect(() => {
+    let source;
+    (async () => {
+      const { CancelToken } = axios;
+      source = CancelToken.source();
+      try {
+        setShowLoading(true);
+        const {
+          data: { data },
+        } = await axios.get(`/api/v1/category/${foodCategoryId}/food`, {
+          cancelToken: source.token,
+        });
+
+        setTotalCalories(
+          (
+            data.reduce(
+              (
+                calories,
+                {
+                  calories_per_gram: caloriesPerGram,
+                  amount_in_grams: amountInGrams,
+                }
+              ) => calories + +caloriesPerGram * +amountInGrams,
+              0
+            ) / 1000
+          ).toFixed(0)
+        );
```

Comment:
This useEffect is pretty long and so that makes it a little hard to quickly understand all of the steps that are happening.  It could help readability if you broke it up into a few smaller functions with descriptive names.  For example, even if you just took the calculation for setTotalCalories and moved it into a helper function, this useEffect would be a lot shorter.